contracts/Marketplace.sol

```diff
@@ -35,7 +46,43 @@ contract Marketplace is Collateral, Proofs {
   }
 
   function myRequests() public view returns (RequestId[] memory) {
-    return _toRequestIds(activeRequests[msg.sender].values());
+    Mappings.ValueId[] storage valueIds =
+      activeClientRequests.getValueIds(Mappings.toKeyId(msg.sender));
+    return _toRequestIds(valueIds);
+  }
+
+  function mySlots()
+    public
+    view
+    returns (SlotId[] memory)
+  {
+    uint256 counter = 0;
+    uint256 totalSlots = activeRequestSlots.getManyCount(); // set this bigger than our possible filtered list size
+    if (totalSlots == 0) {
+      return new SlotId[](0);
+    }
+    bytes32[] memory result = new bytes32[](totalSlots);
```

Comment:
Memory usage of this function will grow with the total amount of slots in the storage network. That could lead to problems down the road.